smallrye-reactive-messaging-aws-sns/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsConnector.java

```diff
@@ -0,0 +1,214 @@
+package io.smallrye.reactive.messaging.aws.sns;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.processors.MulticastProcessor;
+import io.reactivex.schedulers.Schedulers;
+import io.vertx.core.Vertx;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.CreateTopicRequest;
+import software.amazon.awssdk.services.sns.model.CreateTopicResponse;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+
+/**
+ * Implement incoming/outcoming connection factories for SNS reactive messaging.
+ * 
+ * @author iabughosh
+ * @version 1.0.4
+ *
+ */
+@ApplicationScoped
+@Connector(SnsConnector.CONNECTOR_NAME)
+public class SnsConnector implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsConnector.class);
+    static final String CONNECTOR_NAME = "smallrye-aws-sns";
+
+    /*
+     * Injectable fields
+     */
+    @Inject
+    @ConfigProperty(name = "sns-app-public-url")
+    Optional<String> appURL;
+    /** If there is existing verx instance, inject it. */
+    @Inject
+    Instance<Vertx> instanceOfVertx;
+    /*
+     * Fields
+     */
+    /** Control wither close vertx manually or not */
+    private boolean internalVertxInstance = false;
+
+    private Vertx vertx;
+    private ExecutorService threadExecutor;
+    private Scheduler scheduler;
+
+    /**
+     * Method being invoked when CDI bean first created.
+     */
+    @Inject
+    public void initConnector() {
```

Comment:
What does it inject? 
I don't believe it's a valid usage of `@Inject`

Reply:
@Inject on method is like @PostConstruct for CDI beans.Do you want me to switch for @PostConstruct ?